cmd/build/main.go

```diff
@@ -204,32 +82,65 @@ func run() error {
 		return fmt.Errorf("invalid image type %q for distro %q and arch %q: %w", imgTypeName, distroName, archName, err)
 	}
 
-	// get repositories
-	repos, err := loadRepos(repositories, distroName, archName)
-	if err != nil {
-		return fmt.Errorf("failed to get repositories for %s/%s: %w", distroName, archName, err)
+	var reporeg *reporegistry.RepoRegistry
+	var customRepos []rpmmd.RepoConfig
+	if st, err := os.Stat(repositories); err == nil && st.Mode().IsRegular() {
+		repoConfig, err := rpmmd.LoadRepositoriesFromFile(repositories)
+		if err != nil {
+			return fmt.Errorf("failed to load repositories from %q: %w", repositories, err)
+		}
+		customRepos = repoConfig[archName]
+	} else {
+		// HACK: reporegistry hardcodes adding the "repositories"
+		// dir to the "repoConfigPaths" but the interface of
+		// "images" does not expect this.
+		// XXX: should we fix this in reporegistry?
```

Comment:
Yes, we could remove it and adjust all callers, specifically osbuild-composer.

Reply:
Thanks, I added https://github.com/osbuild/images/pull/1179 which does this and also bundles a change that breaks the build so that it is impossible to build unchanged code with this api break (see the comment in there too).